### ISSUE TITLE
Space in project name not marked as invalid

### ISSUE THREAD

**Issue by spoorcc** [dfetch-org/dfetch#287]
A project name should never contain a space. 
But `dfetch validate`, `dfetch check` don't complain.
When running `dfetch update` everything fails.

reported by @sach-edna

**Comment by spoorcc** [comment]
I can't seem to reproduce this issue. Could you provide more details?

**Comment by sach-edna** [comment]
dfetchfile:
```
  - name: jtag logging
    src: Core/EspJtagLog
    dst: Core/Modules/JtagLog
    repo-path: modules/System
    branch: features/esp_jtag_log
```
dfetch check
```
Dfetch (0.6.0)
  jtag logging        : wanted (features/esp_jtag_log), current (features/esp_jtag_log - 83013be8f7f68e0498609cd1e32f3ccfcd8a8dee), available (features/esp_jtag_log)
```
 dfetch validate
```
Dfetch (0.6.0)
  dfetch.yaml         : valid
```
dfetch update jtag logging
```
Dfetch (0.6.0)
Not all projects found! jtag, logging
This manifest contains jtag logging
```

**Comment by spoorcc** [comment]
Now I know what is going on 😄 

The commands all take a list of projects. If a project contains spaces, the name must be delimited with quotes, I tested with the manifest in the example folder:

![image](https://user-images.githubusercontent.com/1345842/162072769-891110f7-05b5-47f0-bde5-5329eabf5825.png)

Maybe I could make the documentation a little more explicit:

```console
$ dfetch update --help
Dfetch (0.6.0)
usage: dfetch update [-h] [-f] [-N] [<project> [<project> ...]]

Update all modules from the manifest.

positional arguments:
  <project>             Specific project(s) to update. If a project name contains spaces, enclose in quotes.

optional arguments:
  -h, --help            show this help message and exit
  -f, --force           Always perform update, ignoring version check or local
                        changes.
  -N, --no-recommendations
                        Ignore recommendations from fetched projects.

Verifies the manifest and checks all dependencies if updates are available.
```

**Comment by spoorcc** [comment]
Fixed in #320
